src/Resources/contao/classes/Frontend.php

```diff
@@ -341,11 +341,9 @@ public static function getRootPageFromUrl()
 				/** @var KernelInterface $kernel */
 				global $kernel;
 
-				$arrParams = array();
-				$arrParams['alias'] = '';
-				$arrParams['_locale'] = $objRootPage->language;
+				$arrParams = array('_locale' => $objRootPage->language);
 
-				$strUrl = $kernel->getContainer()->get('router')->generate('contao_frontend', $arrParams);
+				$strUrl = $kernel->getContainer()->get('router')->generate('contao_root', $arrParams);
```

Comment:
I know what you mean, but that's no longer the case with my PR. Because if `prefix_locale` is enabled, the root always contains a language. There is no `/` route in that case.